### ISSUE TITLE
make_lr_sleuth_object - Error: 'basic_shrink_fun' is not an exported object from 'namespace:sleuth'

### ISSUE THREAD

**Issue by mtietjen27** [warrenmcg/sleuth-ALR#2]
Hi @warrenmcg ,
when I try to make the compositional sleuth object, I get the following error:
```
 alr_so <- make_lr_sleuth_object(sample_to_covariates = yeastS2C, target_mapping = yeastAnnos, aggregation_column = 'gene_id', full_model = ~condition, null_model = ~1, beta = "conditionnoN", denom_name = denom)

Error: 'basic_shrink_fun' is not an exported object from `'namespace:sleuth' 
```

I can't find the origin of this error. Have you come across this before?
Thanks a lot!

**Comment by warrenmcg** [comment]
Hi @mtietjen27,

Thank you again for being a "beta" tester, as you're uncovering all of these little things that did not come up from my end of testing. This is related to a feature that has not been folded into the stable release of sleuth yet that I thought would be.

## Two options
### Option 1: install devel version of sleuth
For the time being, please install my fork of sleuth:
```
devtools::install_github('warrenmcg/sleuth', ref = 'speedy_fit')
```

If you don't want to overwrite your current `sleuth` installation, you can write my forked version to a local library directory. My personal one is `~/test_R_library` but it can be whatever you want
```
my_lib <- '~/test_R_library'
dir.create(my_lib, showWarnings = FALSE) # in case you don't have it created yet

# this makes your local library take priority when installing new packages
# see ?.libPaths for more information
.libPaths(my_lib, .libPaths())
devtools::install_github('warrenmcg/sleuth', ref = 'speedy_fit')

# Now, whenever you load my version of sleuth, make sure you specify the local library
library(sleuth, lib.loc = my_lib)
library(sleuthALR)
```

### Option 2: remove new sleuth API until stable release

I think what I'm going to do for now is remove references to the new sleuth API until it gets folded into the stable release, and so I'm going to move the new API into a "devel" version of sleuth-ALR.

**Comment by warrenmcg** [comment]
Hi @mtietjen27,

I've removed the functionality that caused this error. If you reinstall sleuth-ALR, this error should go away without installing the devel version of sleuth. I expect this to solve the problem, so I'm going to close this issue, but please re-open if it doesn't fix the problem. If a new problem arises, please let me know with a new issue.

Thanks!

**Comment by mtietjen27** [comment]
Thank you @warrenmcg, the error was solved!

**Comment by warrenmcg** [comment]
Note to self, @mtietjen27, and other users:

I found another compatibility issue with `get_lr_functions`, in that the only option for `lr_method` that works with the current sleuth API is `'transform'`; `'both'` will work without an error, but will not work as intended.

To solve both the original issue and the current issue, while still allowing users to use the current version of sleuth, I created a compatibility check and then throw warnings if users attempt to use options that are available in `sleuth` version 0.30.0 (the current stable release). I also made a small version bump on my `speedy_fit` devel version of sleuth (it is now version 0.30.0-1).

So, if users install the stable release of sleuth (0.30.0), they should be able to conduct a normal analysis, and at worst get a warning. If users re-install the current version of `speedy_fit` (version 0.30.0-1) they should be able to take advantage of any options available in `sleuth-ALR`.
